### ISSUE TITLE
Bug: leading semicolon after single-line block = wrong indentation

### ISSUE THREAD

**Issue by 12Me21** [eslint/eslint#15930]
### Environment

Node version: v18.0.0
npm version: v8.6.0
Local ESLint version: Not found
Global ESLint version: v8.16.0 (Currently used)
Operating System: linux 5.10.0-9-amd64

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
	"root": true,
	"env": {"es6": true},
	"rules": {
		"indent": ["error", "tab"],
		"semi": ["warn", "never"],
		"semi-style": ["warn", "first"]
	}
}
```
</details>

```js
if (2 > 1)
	console.log('a')
;[1, 2, 3].forEach(x=>console.log(x))
```

### What did you expect to happen?

(no errors)

### What actually happened?

```
  3:1  error  Expected indentation of 1 tab but found 0  indent
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @12Me21, thanks for the issue!

This does seem like a bug. I'm working on the fix.
